website/static/js/scrollspy.js

```diff
@@ -0,0 +1,44 @@
+// Based in part on
+// https://github.com/makotot/scrollspy/blob/master/src/js/modules/scrollspy.js
+(function () {
+    const OFFSET = 10;
+    let timer;
+    let headingsCache;
+    const findHeadings = () =>
+        document.querySelectorAll('.toc-headings > li > a');
+    const onScroll = () => {
+        if (timer) {  // throttle
+            clearTimeout(timer);
+        }
+        timer = setTimeout(() => {
+            let found = false;
+            let headings = headingsCache ? headingsCache : findHeadings();
+            const scrollTop = window.pageYOffset;
+            // scrollspy uses
+            // document.documentElement.scrollTop || document.body.scrollTop;
+            for (let i = 0; i < headings.length; i++) {
+                // if !found and i is the last element, highlight the last
+                let current = !found;
+                if (!found && i < headings.length - 1) {
+                    const next = headings[i + 1].href.split('#')[1];
+                    const nextHeader = document.getElementById(next);
+                    const top = nextHeader.getBoundingClientRect().top;
+                    current = top > OFFSET;
```

Comment:
This should be `current = top > scrollTop + OFFSET;`

Reply:
It used to be `top = nextHeader.getBoundingClientRect().top + scrollTop` and then `current = top + scrollTop > scrollTop + OFFSET`.  The `scrollTop`s cancel.